tests.sh

```diff
@@ -301,12 +342,22 @@ tests_get_last() {
     cat .last-testcase
 }
 
+tests_set_last() {
+    local testcase=$1
+    echo "$testcase" > .last-testcase
+}
+
 tests_verbose() {
-    export TEST_VERBOSE=$1
+    TEST_VERBOSE=$1
 }
 
 tests_init() {
-    export TEST_ID="$(mktemp -t -d tests.XXXX)"
+    TEST_ID="$(mktemp -t -d tests.XXXX)"
+
+    TEST_STDERR="$TEST_ID/stderr"
```

Comment:
What do you think about keeping stderr/stdout/exitcode separately for each executed command? Should be easier to debug, I guess.